public/script.js

```diff
@@ -0,0 +1,100 @@
+'use strict';
+
+const range = 5;
+const instructionText = `Type a number between 1 and ${range}`;
+
+// STATE
+
+let currentScore;
+let guess = Number(document.querySelector('.guess-input').value);
+let highScore = 0;
+let secretNumber;
+
+// FUNCTIONS
+
+function isValidGuess() {
+  // When there is no input
+  if (!guess) {
+    setTextContent('message', 'Please type a number into the box on the left');
+    return false;
+  }
+  // When guess is not between 1 - (range)
+  if (guess < 1 || guess > range) {
+    setTextContent('message', `Number must be between 1 and ${range}`);
+    return false;
+  }
+  // When input is a decimal
+  if (guess !== Math.trunc(guess)) {
+    setTextContent('message', 'Guess must be a whole number (no decimals)');
+    return false;
+  }
+  return true;
+}
+
+function generateNewSecretNumber() {
+  secretNumber = Math.trunc(Math.random() * range) + 1;
+}
+
+function loseGame() {
+  currentScore = 0;
+  setTextContent('score', currentScore);
+  setTextContent('secret-number', secretNumber);
+  setTextContent('message', 'You lost the game 😱️');
+}
+
+function setTextContent(targetClass, newContent) {
+  return document.querySelector(`.${targetClass}`).textContent = newContent;
+}
+
+function startGame() {
+  generateNewSecretNumber();
+  currentScore = range;
+  setTextContent('secret-number', '?');
+  setTextContent('score', currentScore);
+  setTextContent('message', 'Start guessing...');
+  document.querySelector('.guess-input').value = '';
+  document.querySelector('.secret-number').style.width = '15rem';
+  document.querySelector('body').style.backgroundColor = '#0f0a1e';
+}
+
+function winGame() {
+  setTextContent('secret-number', secretNumber);
+  setTextContent('message', 'You guessed it! 🎉');
+  document.querySelector('body').style.background = '#60b347';
+  document.querySelector('.secret-number').style.width = '30rem';
+  if (currentScore > highScore) {
+    highScore = currentScore;
+    setTextContent('highscore', highScore);
+  }
+}
+
+// INITIALIZE
+
+setTextContent('instructions', instructionText);
+startGame();
+
+// EVENT LISTENERS
+
+document.querySelector('.btn__check').addEventListener('click', function () {
+  guess = Number(document.querySelector('.guess-input').value);
+  if (!isValidGuess()) {
+    return;
+  }
+
+  if (guess === secretNumber) {
+    winGame();
+
+  } else {
+    setTextContent('message', guess > secretNumber ? 'Too high!' : 'Too low!');
+    if (currentScore > 1) {
+      currentScore -= 1;
+      setTextContent('score', currentScore);
+    } else {
+      loseGame();
+    }
```

Comment:
```suggestion
    currentScore = Math.max(currentScore - 1, 0);
    setTextContent('message', guess > secretNumber ? 'Too high!' : 'Too low!');
    setTextContent('score', currentScore);
    if (currentScore === 0) {
      loseGame();
    }
```
Nit: The logic can be simplified a little. The `Math.max()` call prevents the score from dropping below 0.